File: src/1869.longer-contiguous-segments-of-ones-than-zeros.1999/solution.ts
function checkZeroOnes (s: string): boolean {
    let index = 0;
    let maxLength0 = 0;
    let maxLength1 = 0;
    while (index < s.length) {
        let length1 = 0;
        let length0 = 0;
        while (index < s.length && s[index] === '0') {
            length0++;
            index++;
        }
        while (index < s.length && s[index] === '1') {
            length1++;
            index++;
        }
        maxLength0 = Math.max(maxLength0, length0);
        maxLength1 = Math.max(maxLength1, length1);
    }
    return maxLength1 > maxLength0;
}

File: src/1869.longer-contiguous-segments-of-ones-than-zeros.1999/solution.java
class Solution {
    public boolean checkZeroOnes(String s) {
        int max1 = 0;
        int max0 = 0;
        int index = 0;
        char[] chars = s.toCharArray();
        while (index<chars.length){
            char c = chars[index++];
            int count = 1;
            while(index<chars.length && chars[index] == c){
                index++;
                count++;
            }
            if(c == '1'){
                max1 = Math.max(max1,count);
            }else{
                max0 = Math.max(max0,count);
            }
            
        }
        
        return max1>max0;
    }
}